File: c++/advanced/4d_remove_duplicates_st_unique_elements_appers_atmost_twice.cpp
//Given a sorted array nums, remove some duplicates in-place such that each unique element appears at most twice.

int removeDuplicates(vector<int>& nums) {
        int l=0, r=0;
        while(r<nums.size()){
            int count = 1;
            while(r+1<nums.size() && nums[r] == nums[r+1]){
                count++;
                r++;
            }
            int x = min(count, 2);
            while(x){
                nums[l] = nums[r];
                l++;
                x--;
            }
            r++;
        }
        return l;
    }

File: js/advanced/4d_remove_duplicates_st_unique_elements_appers_atmost_twice.js
//Given a sorted array nums, remove some duplicates in-place such that each unique element appears at most twice.

var removeDuplicates = function(nums) {
    let l=0, r=0
    while(r<nums.length){
        let count = 1
        while(r<nums.length-1 && nums[r] == nums[r+1]){
            r++
            count++
        } 
        let x = Math.min(count, 2)
        while(x){
            nums[l] = nums[r]
            l++
            x--
        }
        r++
    }
    return l
};